l10n_ch_account_statement_base_import/models/account_bank_statement_import.py

```diff
@@ -181,3 +181,18 @@ def _create_bank_statement(self, stmt_vals):
                 att = self.env['ir.attachment'].create(att_data)
 
         return statement_id, notifs
+
+    @api.model
+    def _find_bank_account_id(self, account_number):
+        """ Override to find Postfinance account Given IBAN number """
+        bank_account_id = super(AccountBankStatementImport, self).\
+            _find_bank_account_id(account_number)
+        if not bank_account_id and account_number and len(
+                account_number) == 21:
+            pf_formated_acc_number = account_number[9:].lstrip('0')
+            bank_account_ids = self.env['res.partner.bank'].search(
```

Comment:
Please rename this variable `bank_account` you don't have an id but a record set here. Plus this record set only contains one element.